src/js/components/favorite/Favorite.jsx

```diff
@@ -1,73 +1,40 @@
-/**
- * Created by cmeyers on 4/11/16.
- */
 // @flow
 
 import React, { Component, PropTypes } from 'react';
-
-type Props = {
-    checked: boolean,
-    onToggle: (checked: boolean) => void,
-    className: string
-}
+import { Checkbox } from '../forms/Checkbox';
 
 export class Favorite extends Component {
 
-    state: {
-        checked: boolean
-    };
-
     static defaultProps = {
         checked: false,
         className: ''
     };
 
-    componentWillMount() {
-        this._updateState(this.props);
-    }
-
-    componentWillReceiveProps(props: Props) {
-        this._updateState(props);
-    }
-
-    _updateState(props: Props) {
-        this.setState({
-            checked: props.checked
-        });
-    }
-
-    toggle(e: Event) {
-        if (e.target instanceof HTMLInputElement) {
-            const checked = e.target.checked;
-            this.setState({
-                checked: checked
-            });
-
-            if (this.props.onToggle != null){
-                this.props.onToggle(checked);
-            }
-        }
+    get checked() {
+        return this.checkbox.checked;
     }
 
     /* eslint-disable max-len */
     render() {
+        const extraClass = this.props.className || '';
+
         return (
-            <label className={`favorite ${this.props.className}`}
-              onClick={(event) => event.stopPropagation()}
+            <Checkbox
+                ref={checkbox => { this.checkbox = checkbox; }}
+                className={`Favorite ${extraClass}`}
+                checked={this.props.checked}
+                onToggle={this.props.onToggle}
+                label={this.props.label}
             >
-                <input type="checkbox"
-                       onChange={this.toggle.bind(this)}
-                       checked={this.state.checked} />
-
                 <svg className="star-icon" width="288" height="24" viewBox="0 0 288 24" xmlns="http://www.w3.org/2000/svg">
```

Comment:
So define a stateless function component `FavoriteStar` within the file and then drop it Favorite's render method? Sure, I can absolutely do that.